src/System.ValueTuple/src/System/ValueTuple/TupleExtensions.cs

```diff
@@ -0,0 +1,906 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    /// <summary>
+    /// Provides extension methods for <see cref="Tuple"/> instances to interop with C# tuples features (deconstruction syntax, converting from and to <see cref="ValueTuple"/>).
+    /// </summary>
+    public static class TupleExtensions
+    {
+        #region Deconstruct
+        /// <summary>
+        /// Deconstruct a propertly nested <see cref="Tuple"/> with 1 elements.
+        /// </summary>
+        public static void Deconstruct<T1>(
+            this Tuple<T1> value,
+            out T1 item1)
+        {
+            item1 = value.Item1;
```

Comment:
I think NRE is good. Deconstruct is not meant to be called by the user, but it will be called by compiler generated code. So having an ArgumentNullException that reports the argument called 'value' won't help the user much to track down the problem.